### DISCUSSION TITLE
Visualisation of Example Decks

### DISCUSSION THREAD

**Discussion by abigailalexjames** [BLAST-WarpX/warpx#4476] [Data Analysis]
Hi there, 

I'm new to PIC codes and plasma acceleration and I wondered if someone could help me visualise the outputs as I can't see what I'm doing wrong. 

I've managed to install the code and run the 2d example input of a laser driven plasma acceleration from [here](https://github.com/ECP-WarpX/WarpX/blob/development/Examples/Physics_applications/laser_acceleration/inputs_2d). The example code writes outputs as both reduced diagnostics and as a file labelled diags 1. 

I have tried to follow the steps in the documentation and the example output file but nothing works to allow me to read the data. 

1. When trying the simple numpy.genfromtxt to pull from the reduced files I get an error implying a mismatch with expected columns: 
ValueError: Some errors were detected !
    Line 2 (got 12 columns instead of 22)

2. When trying to use from read_raw_data I get 'no module exists' error and I cannot figure out how to access this module. It says that it's part of some post processing package, can this be installed? 

3. I tried using OpenPMD and the path to the diags1 output however I'm not clear if the output is as expected. The example input file says:
```
# Diagnostics
diagnostics.diags_names = diag1
diag1.intervals = 200
diag1.diag_type = Full
diag1.fields_to_plot = Ex Ey Ez Bx By Bz jx jy jz rho

The files I have are: 
diag1000000  diag1000000.old.0672564  diag1000200  diag1000200.old.4467708  reducedfiles
```
When I try `ts = OpenPMDTimeSeries('/path_to_file/diags/diag1000000/')`
```
I get the error: Found no valid files in directory
```

It's really unclear to me how to access the output data and plot it. I have searched the documentation and github and cannot find the answer. I would really appreciate any advice you can give. 

Many thanks,
Abigail

**Comment by ax3l** [comment]
Hi @abigailalexjames,

Cool, very close. The only thing that is missing in your inputs is
```ini
diag1.format = openpmd
```
https://warpx.readthedocs.io/en/latest/usage/parameters.html#full-diagnostics

and then for openPMD-viewer you use the path to the diag1 output folder (which contains all steps of the output series):
```py
ts = OpenPMDTimeSeries('/path_to_file/diags/diag1/')
```
